File: ruby/lib/agentic_commerce_protocol/models/payment_method_type.rb
# frozen_string_literal: true
# This file is @generated
module AgenticCommerceProtocol
  class PaymentMethodType
    CARD = "card".freeze

    def self.all_vars
      @all_vars ||= [CARD].freeze
    end

    def initialize(value)
      unless PaymentMethodType.all_vars.include?(value)
        raise "Invalid ENUM value '#{value}' for class #PaymentMethodType"
      end

      @value = value
    end

    def self.deserialize(value)
      return value if PaymentMethodType.all_vars.include?(value)
      raise "Invalid ENUM value '#{value}' for class #PaymentMethodType"
    end

    def serialize
      @value
    end
  end
end

File: rust/src/models/payment_method_type.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
pub enum PaymentMethodType {
    #[serde(rename = "card")]
    Card,
}

impl fmt::Display for PaymentMethodType {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let value = match self {
            Self::Card => "card",
        };
        f.write_str(value)
    }
}
